rubygems/test/index.test.js

```diff
@@ -0,0 +1,218 @@
+const test = require('ava')
+const sinon = require('sinon')
+const limit = require('call-limit')
+const RubyGemsDependencyResolver = require('../')
+
+test.before((t) => {
+  sinon.stub(limit, 'promise')
+})
+
+test.beforeEach((t) => {
+  t.context.log = { warn: sinon.stub(), error: sinon.stub() }
+  t.context.rubygems = new RubyGemsDependencyResolver({ log: t.context.log })
+  t.context.rubygems.got = sinon.stub()
```

Comment:
nit: i think you can do away with importing `call-limit` and stubbing it, since this line defines the only thing that will actually be called during tests (got)